src/streams.h

```diff
@@ -146,28 +145,29 @@ class VectorReader
      * @param[in]  data Referenced byte vector to overwrite/append
      * @param[in]  pos Starting position. Vector index where reads should start.
      */
-    VectorReader(int type, int version, const std::vector<unsigned char>& data, size_t pos)
-        : m_type(type), m_version(version), m_data(data), m_pos(pos)
+    SpanReader(int type, int version, Span<const unsigned char> data, size_t pos)
+        : m_type(type), m_version(version), m_data(data)
     {
-        if (m_pos > m_data.size()) {
-            throw std::ios_base::failure("VectorReader(...): end of data (m_pos > m_data.size())");
+        if (pos > m_data.size()) {
+            throw std::ios_base::failure("SpanReader(...): end of data (pos > m_data.size())");
         }
+        data = data.subspan(pos);
```

Comment:
Shouldn't this be

```suggestion
       m_data = m_data.subspan(pos);
```

Reply:
Ugh, yes! Will create a PR.

Reply:
Actually, it looks like all call sites use `pos=0`, which makes this harmless. With spans, having a separate `pos` is also pointless as it can be easily done by the caller using `subspan`. I'm just going to drop it.

Reply:
I need it for PSBT things

Reply:
https://github.com/bitcoin/bitcoin/pull/23687